chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -259,9 +258,15 @@ impl NetworkState {
     /// the noncancellable logic will be run in the background anyway.
     fn spawn<R: 'static + Send>(
         &self,
+        description: &'static str,
         fut: impl std::future::Future<Output = R> + 'static + Send,
-    ) -> tokio::task::JoinHandle<R> {
-        self.runtime.handle.spawn(fut.in_current_span())
+    ) -> tokio::sync::oneshot::Receiver<R> {
+        let (tx, rx) = tokio::sync::oneshot::channel();
+        self.ops_spawner.spawn(description, async move {
+            let res = fut.await;
+            let _ = tx.send(res);
+        });
+        rx
```

Comment:
Aah crap, is the only reason why we need to do this because we want to receive a specific return type `R` instead of `()`? I think it might make sense to just modify our own `FutureSpawner` to return an arbit type `R` :(